dao/src/main/java/com/epam/esm/dao/impl/OrderDaoImpl.java

```diff
@@ -52,6 +54,21 @@ public Long getUserOrderCount(Long userId) {
         return entityManager.createQuery(query).getSingleResult();
     }
 
+    @Override
+    public Optional<Order> readUserOrder(Long userId, Long orderId) {
+        CriteriaBuilder criteriaBuilder = entityManager.getCriteriaBuilder();
+        CriteriaQuery<Order> query = criteriaBuilder.createQuery(Order.class);
+        Root<Order> root = query.from(Order.class);
+        Predicate userIdMatch = criteriaBuilder.equal(root.get(ATTRIBUTE_USER_ID), userId);
+        Predicate orderIdMatch = criteriaBuilder.equal(root.get(ATTRIBUTE_ID), orderId);
+        query.select(root).where(criteriaBuilder.and(userIdMatch, orderIdMatch));
+        try {
+            return Optional.of(entityManager.createQuery(query).getSingleResult());
+        } catch (NoResultException e) {
+            return Optional.empty();
+        }

```

Comment:
Suggest you add a message with the text we have no orders instead of the exception